### ISSUE TITLE
Modules for self-compiled Armbian not installed

### ISSUE THREAD

**Issue by Jojo-A** [alexreinert/piVCCU#112]
Hi,
I've set up a new Armbian, which I've compiled by myself. Kernel Headers are installed by the "armbian-config" tool.

``sudo apt install `dpkg --get-selections | grep 'linux-image-' | grep '\sinstall' | sed -e 's/linux-image-\([a-z0-9-]\+\).*/linux-headers-\1/'``
gives
`linux-headers-next-meson64 ist schon die neueste Version (5.60).`

But when I install the pivccu-modules-dpkg. I get
```
pivccu-modules-dkms (1.0.20) wird eingerichtet ...
Loading new pivccu-1.0.20 DKMS files...
dpkg: Warnung: Version »4.18.20-meson64
4.18.8« hat falsche Syntax: ungültiges Zeichen in Revisionsnummer
Building for 4.18.20-meson64
Module build for kernel 4.18.20-meson64 was skipped since the
kernel headers for this kernel does not seem to be installed.
Created symlink /etc/systemd/system/multi-user.target.wants/pivccu-dkms.service \u2192 /lib/systemd/system/pivccu-dkms.service.
pivccu3 (3.41.11-15) wird eingerichtet ...
Created symlink /etc/systemd/system/multi-user.target.wants/pivccu.service \u2192 /lib/systemd/system/pivccu.service.
Job for pivccu.service failed because of unavailable resources or another system error.
See "systemctl status pivccu.service" and "journalctl -xe" for details.
piVCCU: DTB amlogic/meson-gxbb-odroidc2.dtb was already patched
```
Could you please assist me in solving this?

Thanks a ton,
Jojo

**Comment by alexreinert** [comment]
The kernel headers in the apt repository are not matching the installed kernel package as you compiled it by yourself.
You have either to install the kernel package from the apt repo as well or you have to recreate your image with the option, that the headers should be installed, too (somewhere in the build config file)

**Comment by Jojo-A** [comment]
Thanks a lot for your reply!

> You have either to install the kernel package from the apt repo as well

So my first option is to install the headers, which are required by _your_ repo, right? I think this should be 4.18.8... Sadly I don't find this package in my package list. In fact my kernel header package list looks quite strange to me. All package names start with "linux-headers-4.18.0-0.bpo". Nothing with 4.18.8, 4.18.20, generic o_O .

> you have to recreate your image with the option, that the headers should be installed, too

Well, I think "my" kernel headers are installed. I installed them via the armbian-config tool. And in the list of my currently installed packages, I have "linux-headers-next-meson64".

Sorry, when it comes to kernel headers, reposoitories and all this stuff, I am mostly lost... :/ .
What would you suggest to do?

Greetings

**Comment by alexreinert** [comment]
You need to install exactly the headers of the installed kernel. Headers are something like a description where to find things in the compiled kernel. So if you compile the kernel by yourself, there will be no package in any repository, that will match.
And I'm pretty sure, that you have not installed the headers for your kernel. You just installed the headers of the right package name, but not kn the right version. And as you compiled the kernel by yourself, the right version is not available in any repo.
So as said in the first post: Install the kernel from Armbian apt repo that matches to the installed header package or rebuild the image after enabling the install header option in the Armbian build config.

**Comment by Jojo-A** [comment]
Alright, I am compiling a new image with `INSTALL_HEADERS="yes"`. I think to install another kernel won't be a good idea, because I needed an additional driver to be compiled, which was not included in Armbians default build config. So when I install another kernel, this specific driver module won't work anymore, I think...

Thank you for your help! I'll post my results here.

**Comment by Jojo-A** [comment]
Alright,

as you promised, my problem could be fixed in these two ways:
1. switching to another kernel
this was performed by just using the `armbian-config` tool. I switched from version 4.18.20 back to 4.18.8. In my case, this of course  crashed the support of my special driver module, which was compiled fore another kernel.
2. setting up a new Armbian with the latest kernel available, but with headers integrated
I compiled Armbian with kernel version 4.18.20 and set `INSTALL_HEADERS="yes"` ind the config-default.conf.

Thanks again for your help!
